hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmBucketInfo.java

```diff
@@ -615,22 +607,48 @@ public static OmBucketInfo getFromProtobuf(BucketInfo bucketInfo,
         .setQuotaInBytes(bucketInfo.getQuotaInBytes())
         .setUsedNamespace(bucketInfo.getUsedNamespace())
         .setQuotaInNamespace(bucketInfo.getQuotaInNamespace());
-    if (buckLayout != null) {
-      obib.setBucketLayout(buckLayout);
-    } else if (bucketInfo.getBucketLayout() != null) {
-      obib.setBucketLayout(
-          BucketLayout.fromProto(bucketInfo.getBucketLayout()));
-    }
+
     if (bucketInfo.hasDefaultReplicationConfig()) {
-      obib.setDefaultReplicationConfig(
-          DefaultReplicationConfig.fromProto(
-              bucketInfo.getDefaultReplicationConfig()));
+      builder.setDefaultReplicationConfig(
+          DefaultReplicationConfig.fromProto(bucketInfo.getDefaultReplicationConfig()));
     }
     if (bucketInfo.hasObjectID()) {
-      obib.setObjectID(bucketInfo.getObjectID());
+      builder.setObjectID(bucketInfo.getObjectID());
     }
     if (bucketInfo.hasUpdateID()) {
-      obib.setUpdateID(bucketInfo.getUpdateID());
+      builder.setUpdateID(bucketInfo.getUpdateID());
+    }
+    if (bucketInfo.hasSourceVolume()) {
+      builder.setSourceVolume(bucketInfo.getSourceVolume());
+    }
+    if (bucketInfo.hasSourceBucket()) {
+      builder.setSourceBucket(bucketInfo.getSourceBucket());
+    }
+    if (bucketInfo.hasOwner()) {
+      builder.setOwner(bucketInfo.getOwner());
+    }
+    if (bucketInfo.hasBucketLayout()) {
+      builder.setBucketLayout(BucketLayout.fromProto(bucketInfo.getBucketLayout()));
+    }
+
+    return builder;
+  }
+
+  /**
+   * Parses BucketInfo protobuf and creates OmBucketInfo.
+   * @param bucketInfo
+   * @return instance of OmBucketInfo
+   */
```

Comment:
This javadoce does not provide any useful information and generate "tag description is missing" warning.  Please remove it or rewrite it.